deviceModeInit.js

```diff
@@ -62,28 +62,69 @@ var rudderTracking = (function () {
     { dest: "url", src: "url" },
   ];
 
+
   function init() {
     pageCurrency = Shopify.currency.active;
     userId = ShopifyAnalytics.meta.page.customerId || __st.cid;
+
+    // fetching heap Cookie object
+    // TODO: for adding dynamic support from source config
+    heapCookieObject = cookie_action({ action: "get", name: "_hp2_id.1200528076"});
+    
+    if (heapCookieObject) {
+      heapCookieObject = JSON.parse(decodeURIComponent(heapCookieObject));
+    } else {
+      console.log("No heap cookie found.")
+    }
+
     htmlSelector.buttonAddToCart =
       rs$('form[action="/cart/add"] [type="submit"]').length === 1
         ? rs$('form[action="/cart/add"] [type="submit"]')
         : "";
-    if (
-      userId &&
-      cookie_action({ action: "get", name: "rudder_user_id" }) !== "captured"
-    ) {
-      rudderanalytics.identify(String(userId));
+        
+    identifyUser()
+  
+    trackPageEvent();
+    trackNamedPageView();
+
+    rs$("button[data-search-form-submit]").on("click", trackProductSearch);
+  }
+  function identifyUser() {
+    if(userId && cookie_action({ action: "get", name: "rudder_user_id" }) !== "captured") {
+      
+      if (heapCookieObject) {
```

Comment:
```suggestion
      if (heapCookieObject && cookie_action({ action: "get", name: "rudder_heap_identities" }) !== "captured") {
```

Reply:
Done. Addressed typos in heapUserId and heapSessionId